torch/autograd/function.py

```diff
@@ -533,12 +534,29 @@ def vmap(info, in_dims, *args):
 
     @classmethod
     def apply(cls, *args, **kwargs):
+        def bind_default_args(func, is_def_ctx, *args, **kwargs):
+            arguments: List[Any]
+
+            arguments = list(args)
+            # Used to bind to ctx
+            if is_def_ctx:
+                arguments.insert(0, None)
+
+            signature = inspect.signature(func)
+            bound_args = signature.bind(*arguments, **kwargs)
+            bound_args.apply_defaults()
+
+            return bound_args.args[1:] if is_def_ctx else bound_args.args
+
+        is_def_ctx = cls.setup_context == _SingleLevelFunction.setup_context
+
         if not torch._C._are_functorch_transforms_active():
             # See NOTE: [functorch vjp and autograd interaction]
             args = _functorch.utils.unwrap_dead_wrappers(args)
+            args = bind_default_args(cls.forward, is_def_ctx, *args, **kwargs)
```

Comment:
We probably also want this for the functorch path? Most usages with the setup context are probably with functorch.

Reply:
I'm not familiar with functorch so I might need some time to learn it and will submit a new PR to fix it tomorrow

Reply:
After briefly browsing the implementation of functorch and doing some simple tests, Intuitively, I believe that functorch should not require additional changes. Since I'm really new to this, if I'm missing something please let me know, thanks :D

Reply:
Sorry, functorch alse need this, I have submitted a new commit, which is ready for review, please have a look at this, thank you.